src/cfulist.c

```diff
@@ -412,6 +412,28 @@ cfulist_pop(cfulist_t *list) {
 	return NULL;
 }
 
+void
+cfulist_delete_data(cfulist_t *list, void *data) {
+	cfulist_entry *ptr = NULL;
+
+	if (!list) {
+		return;
+	}
+
+	lock_list(list);
+
+	if (list->entries) {
+		for (ptr = list->entries; ptr && ptr->data != data; ptr = ptr->next)
+		;
+		if (ptr && ptr->data == data) {
```

Comment:
Does this work for the first item? I didn't look at the implementation of the list very well, but usually the first element of a doubly linked list's `prev` pointer is `NULL`, in which case this looks like a crasher, no?

Reply:
Yes, it was a crasher. I have now pushed a fix. I think that now it really is ready to merge ._.